.github/workflows/connector_checklist.yml

```diff
@@ -0,0 +1,23 @@
+name: Add Connector Merge Checklist
+on:
+  pull_request:
+    types: [opened]
+    paths:
+      - "airbyte-integrations/connectors/source-**"
+      - "airbyte-integrations/connectors/destination-**"
```

Comment:
Do we want to add `airbyte-integrations/connectors/third-party/`? (Syntax might not be right for referencing the folder)